Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs

```diff
@@ -0,0 +1,365 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Serialization;
+
+using CsvHelper;
+
+using Microsoft.CodeAnalysis.Sarif;
+
+using Newtonsoft.Json;
+
+using Taxonomy.Common;
+
+namespace Taxonomy.Cwe
+{
+    public class CweTaxonomyGenerator : TaxonomyGenerator
+    {
+        public bool SaveCsvToSarif(string filePath, string version, string releaseDateUtc)
+        {
+            try
+            {
+                List<CweCsvRecord> records = this.ReadFromCsv(filePath);
+
+                Run run = this.ConvertToSarif(records, version, releaseDateUtc);
+
+                SarifLog log = new SarifLog
+                {
+                    Runs = new Run[] { run }
+                };
+
+                File.WriteAllText(filePath + ".sarif", JsonConvert.SerializeObject(log, Newtonsoft.Json.Formatting.Indented));
```

Comment:
>filePath + ".sarif" [](http://example.com/codeflow?start=34&length=19)

base on my understanding, we should have a string galled targetFilePath, which already ends with .sarif